packages/utils/src/grid/navigation.ts

```diff
@@ -0,0 +1,190 @@
+  const row = grid[rowIndex];
+  if (!row) {
+    return null;
+  }
+
+  for (let i = 0; i < row.length; i += 1) {
+    const nextColumnIndex = target + (i * direction);
+    if (row[nextColumnIndex]) {
+      return [nextColumnIndex, rowIndex];
+    }
+    const prevColumnIndex = target - (i * direction);
+    if (row[prevColumnIndex]) {
+      return [prevColumnIndex, rowIndex];
+    }
+  }
+
+  return null;
+};
+
+/**
+ * Navigate left from the start cell.
+ * If there is no active cell on the left,
+ * than iterate over preceding rows to to find one
+ * @param grid Navigation grid
+ * @param cell Start cell
+ * @returns cell Previous cell. If none return null
+ */
+const left = (grid: NavigationGrid, cell: CellIndex): CellIndex | null => {
+  let [columnIndex, rowIndex] = cell;
+  const [columnCount] = counts(grid);
+
+  while (rowIndex >= 0) {
+    columnIndex -= 1;
+    if (columnIndex < 0) {
+      rowIndex -= 1;
+      columnIndex = columnCount;
+    }
+    const row = grid[rowIndex];
+    if (!row) {
+      break;
+    }
+    const cell = row[columnIndex];
+    if (cell) {
+      return [columnIndex, rowIndex];
+    }
+  }
+
+  return null;
+};
+
+/**
+ * Navigate right from the start cell.
+ * If there is no active cell on the right,
+ * than iterate over following rows to to find one
+ * @param grid Navigation grid
+ * @param cell Start cell
+ * @returns cell Next cell. If none return null
+ */
+const right = (grid: NavigationGrid, cell: CellIndex): CellIndex | null => {
+  let [columnIndex, rowIndex] = cell;
+  const [columnCount, rowCount] = counts(grid);
+
+  while (rowIndex < rowCount) {
+    columnIndex += 1;
+    if (columnIndex >= columnCount) {
+      rowIndex += 1;
+      columnIndex = 0;
+    }
+
+    const row = grid[rowIndex];
+    if (!row) {
+      break;
+    }
+    const cell = row[columnIndex];
+    if (cell) {
+      return [columnIndex, rowIndex];
+    }
+  }
+
+  return null;
+};
+
+/**
+ * Navigate up from the start cell trying to
+ * find the closest cell on the preceding rows
+ * @param grid Navigation grid
+ * @param cell Start cell
+ * @returns cell Closest cell on the previous row. If none return null
+ */
+const up = (grid: NavigationGrid, cell: CellIndex): CellIndex | null => {
+  const columnIndex = cell[0];
+  let rowIndex = cell[1];
+
+  while ((rowIndex -= 1) >= 0) {
+    const cell = closest(grid, rowIndex, columnIndex, 1);
+    if (cell) {
+      return cell;
+    }
+  }
+
+  return null;
+};
+
+/**
+ * Navigate down from the start cell trying to
+ * find the closest cell on the following rows
+ * @param grid Navigation grid
+ * @param cell Start cell
+ * @returns cell Closest cell on the next row. If none return null
+ */
+const down = (grid: NavigationGrid, cell: CellIndex): CellIndex | null => {
+  const columnIndex = cell[0];
+  let rowIndex = cell[1];
+  const rowCount = counts(grid)[1];
```

Comment:
If you want to keep it the same as the others, I think you can do
```suggestion
  const [,rowCount] = counts(grid);
```

Reply:
Done